src/collects/seashell/backend/files.rkt

```diff
@@ -223,3 +225,47 @@
       (with-input-from-file (build-path (read-config 'seashell) "settings.txt")
         (thunk (read)))]
     [else #f]))
+
+;; (update-recent-file project file)
+;; Updates ~/.seashell/recent.txt to say that `file' was the most
+;; recent file for the project `project'.
+;;
+;; Arguments:
+;;  project - the project to update.
+;;  file - the most recent file.
+(define/contract (update-recent-file project file)
+  (-> (and/c project-name? is-project?) path-string? void?)
+  (define recent (build-path (read-config 'seashell) "recent.txt"))
+  (when (not (file-exists? recent))
+    (with-output-to-file recent (thunk (write (make-hash)))))
+  (define r (with-input-from-file recent read))
+
+  (check-and-build-path project file)   ;Make sure this is a real
+                                        ;project file.
+  (define (update-tree f)
+    (define-values (base name _) (split-path f))
+    (define dir (if (path? base)
+                    (string-append project "/" (path->string base))
+                    project))
+    (hash-set! r dir name)
+    (cond
+     [(path? base) (update-tree base)]
+     [else (hash-set! r project name)]))
+
+  (update-tree file)
+
+  (with-output-to-file recent (thunk (write r)))
+  (void))
+
+;; (get-recent-file directory)
+;; Reads the most recent file name for the specified project.
+;;
+;; Arguments:
+;;  directory - the directory to check the default for.
+(define/contract (get-recent-file directory)
```

Comment:
One final thing; get-recent-file should take in a project, subdirectory pair for consistency.